### PR TITLE
Update citation in pvlib readme

### PR DESCRIPTION
**Pull request by AdamRJensen**
- [ ] Closes #xxxx
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] Tests added
 - [ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

A new paper concerning pvlib has been published in JOSS: https://doi.org/10.21105/joss.05994

Therefore, it seems fitting to update the recommended citation. I might make sense to keep a reference to the 2018 JOSS paper somewhere - ideas?

### CODE REVIEW

**Comment by AdamRJensen** [comment]
The CITATION.cff file should probably also be updated

File: README.md [hunk 1]
```diff
@@ -101,15 +101,21 @@ Citing
 
 If you use pvlib-python in a published work, please cite:
 
-  William F. Holmgren, Clifford W. Hansen, and Mark A. Mikofski.
-  "pvlib python: a python package for modeling solar energy systems."
-  Journal of Open Source Software, 3(29), 884, (2018).
-  https://doi.org/10.21105/joss.00884
+  Anderson et al., (2023). pvlib python: 2023 project update.
```
Comment (kandersolar) [inline README.md:104]: Could we list all the authors instead of using et al?  Maybe just last names, to save space.

File: README.md [hunk 2]
```diff
@@ -101,15 +101,21 @@ Citing
 
 If you use pvlib-python in a published work, please cite:
 
-  William F. Holmgren, Clifford W. Hansen, and Mark A. Mikofski.
-  "pvlib python: a python package for modeling solar energy systems."
-  Journal of Open Source Software, 3(29), 884, (2018).
-  https://doi.org/10.21105/joss.00884
```
Comment (kandersolar) [inline README.md:107]: Could we keep the 2018 citation too?  Maybe a category structure like this, which leaves room for future expansion:

```
If you use pvlib-python in a published work, please cite as appropriate:

Project as a whole:
- 2023 JOSS citation
- 2018 JOSS citation

Data retrieval using `pvlib.iotools`:
- 2023 iotools citation
```

**Review by kandersolar — COMMENTED** [review]
> The CITATION.cff file should probably also be updated

We haven't got one, see https://github.com/pvlib/pvlib-python/issues/1322.

Also, as long as we're editing the README, can you fix the link in the "history and acknowledgement" section?  I copy/pasted RST syntax instead of markdown by accident.

File: README.md (hunk 1 above)
Comment (AdamRJensen) [reply README.md:104]: Certainly

Comment (AdamRJensen) [reply README.md:104]: Has now been done. Let me know if you don't like the style of "Last name, First letter of first name"

File: docs/sphinx/source/index.rst [hunk 3]
```diff
@@ -48,28 +48,37 @@ companies that allow their employees to contribute, and from personal time.
...
+  Anderson, K., Hansen, C., Holmgren, W., Jensen, A., Mikofski, M., and Driesse, A.
+  "pvlib python: 2023 project update."
+  Journal of Open Source Software, 8(92), 5994, (2023).
+  https://doi.org/10.21105/joss.05994
+
+
+**Recommended citation for pvlib iotools**
+
+  Jensen, A., Anderson, K., Holmgren, W., Mikofski, M., Hansen, C., Boeman, L., Loonen, R.
+  "pvlib iotools —- Open-source Python functions for seamless access to solar irradiance data."
```
Comment (kandersolar) [inline docs/sphinx/source/index.rst:66]:
```suggestion
  "pvlib iotools — Open-source Python functions for seamless access to solar irradiance data."
```

File: docs/sphinx/source/index.rst [hunk 4]
```diff
@@ -107,22 +129,3 @@ pvlib python is a `NumFOCUS Affiliated Project <https://numfocus.org/sponsored-p
 .. image:: https://i0.wp.com/numfocus.org/wp-content/uploads/2019/06/AffiliatedProject.png
   :target: https://numfocus.org/sponsored-projects/affiliated-projects
   :alt: NumFocus Affliated Projects
-
-Contents
-========
-
-.. toctree::
-   :maxdepth: 1
```
Comment (kandersolar) [inline docs/sphinx/source/index.rst:115]: This toctree is what defines the whole documentation website structure, so we better not delete it :P

You can add `:hidden:` if you don't want it to be displayed in the page content.

File: README.md (hunk 1 above)
Comment (kandersolar) [reply README.md:104]: Can we do it here in README.md too?  Thanks :)

**Review by cwhanse — APPROVED** [review]

**Comment by AdamRJensen** [comment]
@pvlib/pvlib-core Merging this tomorrow if no objections

**Pull request merged**